eos-updater-flatpak-installer/eos-updater-flatpak-installer.c

```diff
@@ -0,0 +1,262 @@
+        }
+
+      if (!parse_flatpak_ref_kind (ref_kind, &parsed_ref_kind))
+        {
+          malformed_flatpak_to_deploy_file_error (error, *split_contents_iter);
+          return NULL;
+        }
+
+      g_ptr_array_add (flatpaks_to_deploy,
+                       flatpak_to_deploy_new (repo, app_id, parsed_ref_kind));
+    }
+
+  return g_steal_pointer (&flatpaks_to_deploy);
+}
+
+static GPtrArray *
+get_flatpaks_to_deploy (const gchar *to_deploy_file_path, GError **error)
+{
+  g_autofree gchar *contents = NULL;
+
+  if (!g_file_get_contents (to_deploy_file_path, &contents, NULL, error))
+    return NULL;
+
+  return parse_flatpaks_to_deploy_from_contents (contents, error);
+}
+
+static const gchar *
+pending_flatpak_deployments_state_path (void)
+{
+  return eos_updater_get_envvar_or ("EOS_UPDATER_TEST_UPDATER_FLATPAK_UPGRADE_STATE_DIR",
+                                    LOCALSTATEDIR "/lib/eos-updater");
+}
+
+static char *
+format_flatpaks (GPtrArray *flatpaks_to_deploy)
+{
+  g_auto(GStrv) flatpaks_to_deploy_strv = g_new0 (gchar *, flatpaks_to_deploy->len + 1);
+  gsize i = 0;
+
+  for (; i < flatpaks_to_deploy->len; ++i)
+    flatpaks_to_deploy_strv[i] = ((FlatpakToDeploy *) g_ptr_array_index (flatpaks_to_deploy, i))->app_id;
+
+  return g_strjoinv (", ", flatpaks_to_deploy_strv);
+}
+
+static FlatpakInstallation *
+get_flatpak_installation (GError **error)
+{
+  const gchar *override_path = g_getenv ("EOS_UPDATER_TEST_FLATPAK_INSTALLATION_DIR");
+
+  if (override_path)
+    {
+      g_autoptr(GFile) override = g_file_new_for_path (override_path);
+      return flatpak_installation_new_for_path (override, TRUE, NULL, error);
+    }
+
+  return flatpak_installation_new_system (NULL, error);
+}
+
+static gboolean
+deploy_flatpaks (GPtrArray *flatpaks_to_deploy, GError **error)
+{
+  g_autoptr(FlatpakInstallation) installation = get_flatpak_installation (error);
+  gsize i = 0;
+
+  if (!installation)
+    return FALSE;
+
+  for (; i < flatpaks_to_deploy->len; ++i)
+    {
+      FlatpakToDeploy *to_deploy = (FlatpakToDeploy *) g_ptr_array_index (flatpaks_to_deploy,
+                                                                         i);
+      if (!flatpak_installation_install_full (installation,
+                                              FLATPAK_INSTALL_FLAGS_NO_PULL,
+                                              to_deploy->repo,
+                                              to_deploy->ref_kind,
+                                              to_deploy->app_id,
+                                              NULL,
+                                              NULL,
+                                              NULL,
+                                              NULL,
+                                              NULL,
+                                              NULL,
+                                              error))
+        return FALSE;
+    }
+
+  return TRUE;
+}
+
+int
+main (int    argc,
```

Comment:
Also, I think this should have a `GOptionContext` so that running it with `--help` does what the user might expect.